### ISSUE TITLE
Dr. Memory internal crash at PC 0x6e63d99b

### ISSUE THREAD

**Issue by lefty2** [DynamoRIO/drmemory#2319]
When I try to use dr. memory on Windows 10 I get this error:

Dr. Memory version 2.3.0
         Running "c:\backup\warlord\Warlord\debug\MercTactics.exe"
         Using system call file C:\Users\Lefty\AppData\Roaming\Dr. Memory\symcache\syscalls_wow64.txt
<Application c:\backup\warlord\Warlord\debug\MercTactics.exe (868).  Dr. Memory internal crash at PC 0x6e63d99b.  Please report this at http://drmemory.org/issues along with the results of running '-debug -dr_debug'.  Program aborted.
0xc0000005 0x00000000 0x6e63d99b 0x6e63d99b 0x00000003 0x00000000
Base: 0x6e590000
Registers: eax=0x00000000 ebx=0x0019f688 ecx=0x0883de54 edx=0x00000000
	esi=0x17dc6b90 edi=0x17dc6b88 esp=0x0019f6a8 ebp=0x0019f75c
	eflags=0x000
2.3.0-1-(Feb  6 2020 06:09:03) WinVer=105;Rel=1903;Build=18362;Edition=Enterprise
-no_dynamic_options -disasm_mask 8 -logdir 'C:\Users\Lefty\AppData\Roaming\Dr. Memory\dynamorio' -client_lib 'C:\Program Files (x86)\Dr. Memory\bin\release\drmemorylib.dll;0;-visual_studio -logdir `C:\Users\Lefty\AppData\Roaming\Dr. Memory` -symcache_dir `C:\Users\Lefty\AppData\Roaming\Dr. Memory\symcache` -lib_blacklist
0x0019f75c 0x6e616460>
         WARNING: application exited with abnormal code 0xffffffff

Dr. Memory version 2.3.0
Running "c:\backup\warlord\Warlord\debug\MercTactics.exe"
Using system call file C:\Users\Lefty\AppData\Roaming\Dr. Memory\symcache\syscalls_wow64.txt
<Application c:\backup\warlord\Warlord\debug\MercTactics.exe (11796). Dr. Memory internal crash at PC 0x6f86d99b. Please report this at http://drmemory.org/issues along with the results of running '-debug -dr_debug'. Program aborted.
0xc0000005 0x00000000 0x6f86d99b 0x6f86d99b 0x00000003 0x00000000
Base: 0x6f7c0000
Registers: eax=0x00000000 ebx=0x0019f688 ecx=0x26c1de54 edx=0x00000000
esi=0x1a116b90 edi=0x1a116b88 esp=0x0019f6a8 ebp=0x0019f75c
eflags=0x000
2.3.0-1-(Feb 6 2020 06:09:03) WinVer=105;Rel=1903;Build=18362;Edition=Enterprise
-no_dynamic_options -disasm_mask 8 -logdir 'C:\Users\Lefty\AppData\Roaming\Dr. Memory\dynamorio' -client_lib 'C:\Program Files (x86)\Dr. Memory\bin\release\drmemorylib.dll;0;-visual_studio -logdir C:\Users\Lefty\AppData\Roaming\Dr. Memory -symcache_dir C:\Users\Lefty\AppData\Roaming\Dr. Memory\symcache -lib_blacklist
0x0019f75c 0x6f846460>
WARNING: application exited with abnormal code 0xffffffff

Application c:\backup\warlord\Warlord\debug\MercTactics.exe (11796). Dr. Memory internal crash at PC 0x6f86d99b. Please report this at http://drmemory.org/issues along with the results of running '-debug -dr_debug'. Program aborted.
0xc0000005 0x00000000 0x6f86d99b 0x6f86d99b 0x00000003 0x00000000
Base: 0x6f7c0000
Registers: eax=0x00000000 ebx=0x0019f688 ecx=0x26c1de54 edx=0x00000000
esi=0x1a116b90 edi=0x1a116b88 esp=0x0019f6a8 ebp=0x0019f75c
eflags=0x000
2.3.0-1-(Feb 6 2020 06:09:03) WinVer=105;Rel=1903;Build=18362;Edition=Enterprise
-no_dynamic_options -disasm_mask 8 -logdir 'C:\Users\Lefty\AppData\Roaming\Dr. Memory\dynamorio' -client_lib 'C:\Program Files (x86)\Dr. Memory\bin\release\drmemorylib.dll;0;-visual_studio -logdir C:\Users\Lefty\AppData\Roaming\Dr. Memory -symcache_dir C:\Users\Lefty\AppData\Roaming\Dr. Memory\symcache -lib_blacklist
0x0019f75c 0x6f846460

**Comment by derekbruening** [comment]
As already requested (!!!  https://github.com/DynamoRIO/drmemory/issues/2232#issuecomment-712485938) please fill out the bug template.  Without more information this report is not actionable.

**Comment by lefty2** [comment]
I ran with all the below options and it crashes just the same way with the same error messages above.

- Does the problem go away when running in light mode (pass `-light` to Dr. Memory)?
 - Does the problem go away when running with the options `-leaks_only -no_count_leaks -no_track_allocs`?
 - Does the problem go away when running under plain DynamoRIO?  Do this by running `dynamorio/bin32/drrun -- <application and args>` or `dynamorio/bin64/drrun -- <application and args>` depending on the bitwidth of your applicaiton.  (Ignore warnings about "incomplete installation".)
 - What happens with the debug version of Dr. Memory and of its underlying engine DynamoRIO?  Try this by passing `-debug -dr_debug -pause_at_assert`.  Are any messages reported?

**Comment by lefty2** [comment]
I'll attach the executable to you can try for yourself
[http://astronautz.com/MercTactics/debug.7z](url)

**Comment by derekbruening** [comment]
> I ran with all the below options and it crashes just the same way with the same error messages above.

That is impossible: `dynamorio/bin32/drrun` is not Dr. Memory and cannot produce the same message.

Given no other provided info, we have to assume that all other applications work fine on this same machine, and that this is something that only happens with this particular "MercTactics" executable.

**Comment by lefty2** [comment]
This is the exact message that drrun gives:
---------------------------
DynamoRIO Notice: C:\backup\warlord\Warlord\debug\MercTactics.exe(8844)
---------------------------
Application C:\backup\warlord\Warlord\debug\MercTactics.exe (8844).  DynamoRIO internal crash at PC 0x6e63d99b.  Please report this at http://dynamorio.org/issues/.  Program aborted.
0xc0000005 0x00000000 0x6e63d99b 0x6e63d99b 0x00000003 0x00000000
Base: 0x6e590000
Registers: eax=0x00000000 ebx=0x0019f688 ecx=0x0883de54 edx=0x00000000
	esi=0x199240a0 edi=0x19924098 esp=0x0019f6a8 ebp=0x0019f75c
	eflags=0x0001
version 7.91.18298, custom build
-no_dynamic_options -code_api -probe_api -stack_size 56K -max_elide_jmp 0 -max_elide_call 0 -no_inline_ignored_syscalls -native_exec_default_list '' -no_native_exec_managed_code -no_indcall2direct -no_aslr_dr 
0x0019f75c 0x6e616460

I also tried running on an application compiled with GCC and it crashes the same way.
Dr. memory runs this application fine on Windows 8.1, so it's looks more like this version of Windows 10 that's causing the problems

**Comment by lefty2** [comment]
> > I ran with all the below options and it crashes just the same way with the same error messages above.
> 
> That is impossible: `dynamorio/bin32/drrun` is not Dr. Memory and cannot produce the same message.
> 
> Given no other provided info, we have to assume that all other applications work fine on this same machine, and that this is something that only happens with this particular "MercTactics" executable.
This is not true. I tried with several other applications and they all have the same problem

**Comment by derekbruening** [comment]
> > > I ran with all the below options and it crashes just the same way with the same error messages above.
> > 
> > 
> > That is impossible: `dynamorio/bin32/drrun` is not Dr. Memory and cannot produce the same message.
> > Given no other provided info, we have to assume that all other applications work fine on this same machine, and that this is something that only happens with this particular "MercTactics" executable.
> 
> This is not true. I tried with several other applications and they all have the same problem

But why file an issue on a large complex executable "MercTactics" if it reproduces with a tiny hello,world executable?  Always minimize before filing, and describe the context in the filing.

If hello,world crashes under plain DR, there is likely some invasive software on this particular machine causing interoperability problems.  Certainly simple apps work on Windows 10 1903 on other machines.  Most likely this requires attaching the debugger on the machine where this reproduces and getting a callstack, and looking at what security or anti-virus or other software may be causing the problem by injecting libraries into every process that hook or otherwise clobber code.

**Comment by lefty2** [comment]
I see a lot of similar crash bugs filed here, so it appears it's not just my app that has a problem. I disabled the Anti virus. 
I don't know how you come to the conclusion that something is "injecting libraries into every process that hook or otherwise clobber code", this is just a vanilla install of Windows 10.
